test/unit/models_test.rb

```diff
@@ -21,7 +21,8 @@ class ModelsTest < ActiveSupport::TestCase
       'LatestForumPostsPortlet' => %w[content_type], 'TableOfContentsPortlet' => %w[content_type], 'AuthenticationProvider::GitHub' => %w[branding_state account_type],
       'AuthenticationProvider::Keycloak' => %w[account_type], 'AuthenticationProvider::Auth0' => %w[account_type], 'AuthenticationProvider::Custom' => %w[account_type],
       'AuthenticationProvider::ServiceDiscoveryProvider' => %w[account_type], 'AuthenticationProvider::RedhatCustomerPortal' => %w[account_type],
-      'Account' => %w[credit_card_auth_code credit_card_authorize_net_payment_profile_token credit_card_partial_number]
+      'Account' => %w[credit_card_auth_code credit_card_authorize_net_payment_profile_token credit_card_partial_number],
+      'Annotation' => %w[annotated_type], 'Plan' => %w[issuer_type], 'ApplicationPlan' => %w[issuer_type], 'ServicePlan' => %w[issuer_type], 'AccountPlan' => %w[issuer_type]
```

Comment:
> Also I think it is better to keep each class on a separate line. I see some previous ones were put on the same line but maybe because they considered them related. IDK, not really easy to look at them like this IMO.

Yeah, there are a bunch of models that are merged on one line, I don't think they are related, but I think they are just short (one attribute).

While I agree that it's hard to read, this is also just a single test, so you wouldn't need to look at it very often - unless you change this specific test, or upgrade Rails or something like that :) But yeah, we can split them and maybe order alphabetically for a better readability.

Reply:
> Why excluding the `whatever_type` fields?

IIRC, the issue was that these polymorphic associations were problematic due to the following error:
```
Minitest::UnexpectedError: NameError: wrong constant name aa[....]aaaa

      Object.const_get(camel_cased_word)
```

So, if the value of such `whatever_type` field reaches the max length for the field, then Rails was trying to get the constant for this class, and failed.

Full trace:
```
wrong constant name aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa

      Object.const_get(camel_cased_word)
            ^^^^^^^^^^
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.8.6/lib/active_support/inflector/methods.rb:278:in `const_get'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.8.6/lib/active_support/inflector/methods.rb:278:in `constantize'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.8.6/lib/active_support/core_ext/string/inflections.rb:74:in `constantize'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activerecord-7.0.8.6/lib/active_record/inheritance.rb:207:in `polymorphic_class_for'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activerecord-7.0.8.6/lib/active_record/associations/belongs_to_polymorphic_association.rb:9:in `klass'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activerecord-7.0.8.6/lib/active_record/associations/association.rb:215:in `ensure_klass_exists!'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activerecord-7.0.8.6/lib/active_record/associations/singular_association.rb:8:in `reader'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/bullet-7.0.7/lib/bullet/active_record70.rb:234:in `reader'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activerecord-7.0.8.6/lib/active_record/associations/builder/association.rb:104:in `annotated'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activemodel-7.0.8.6/lib/active_model/validator.rb:150:in `block in validate'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activemodel-7.0.8.6/lib/active_model/validator.rb:149:in `each'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activemodel-7.0.8.6/lib/active_model/validator.rb:149:in `validate'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.8.6/lib/active_support/callbacks.rb:423:in `block in make_lambda'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.8.6/lib/active_support/callbacks.rb:199:in `block (2 levels) in halting'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.8.6/lib/active_support/callbacks.rb:687:in `block (2 levels) in default_terminator'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.8.6/lib/active_support/callbacks.rb:686:in `catch'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.8.6/lib/active_support/callbacks.rb:686:in `block in default_terminator'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.8.6/lib/active_support/callbacks.rb:200:in `block in halting'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.8.6/lib/active_support/callbacks.rb:595:in `block in invoke_before'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.8.6/lib/active_support/callbacks.rb:595:in `each'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.8.6/lib/active_support/callbacks.rb:595:in `invoke_before'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.8.6/lib/active_support/callbacks.rb:106:in `run_callbacks'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.8.6/lib/active_support/callbacks.rb:929:in `_run_validate_callbacks'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activemodel-7.0.8.6/lib/active_model/validations.rb:406:in `run_validations!'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activemodel-7.0.8.6/lib/active_model/validations/callbacks.rb:115:in `block in run_validations!'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.8.6/lib/active_support/callbacks.rb:99:in `run_callbacks'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activesupport-7.0.8.6/lib/active_support/callbacks.rb:929:in `_run_validation_callbacks'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activemodel-7.0.8.6/lib/active_model/validations/callbacks.rb:115:in `run_validations!'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activemodel-7.0.8.6/lib/active_model/validations.rb:337:in `valid?'
/opt/ci/workdir/vendor/bundle/ruby/3.1.0/gems/activerecord-7.0.8.6/lib/active_record/validations.rb:68:in `valid?'
/opt/ci/workdir/test/unit/models_test.rb:45:in `block (3 levels) in <class:ModelsTest>'
/opt/ci/workdir/test/unit/models_test.rb:36:in `each'
/opt/ci/workdir/test/unit/models_test.rb:36:in `block (2 levels) in <class:ModelsTest>'
/opt/ci/workdir/test/unit/models_test.rb:54:in `block (2 levels) in <class:ModelsTest>'
/opt/ci/workdir/test/unit/models_test.rb:53:in `each'
/opt/ci/workdir/test/unit/models_test.rb:53:in `block in <class:ModelsTest>'
```

Reply:
Ok. The test is a little useless for these fields anyway. Just if we add a class with polymorphic association or with STI, the test will have to be updated. But I assume we add classes not more often than upgrading rails so might be acceptable.